### ISSUE TITLE
Invalid Parameter for Block device with Snapshot ID -- Terraform v0.5.0

### ISSUE THREAD

**Issue by randylyon** [hashicorp/terraform#1862]
Receiving this error when using Terraform 0.5.0 with a block device specifying a snapshot.  I get this error if I add encryption = false or don't have encryption defined at all.

* Error launching source instance: InvalidParameter: Parameter encrypted is invalid. You cannot specify the encrypted flag if specifying a snapshot id in a block device mapping.

Instance defined as:

```
resource "aws_instance" "test" {
    ami = "${var.instance_ami}"
    instance_type = "t2.medium"
    key_name = "REDACTED"
    availability_zone = "us-east-1a"
    user_data = "${file(\"userData.txt\")}"
    tags {
        Name = "test"
    }
    associate_public_ip_address = true
    security_groups = ["Web-Application-Default"]
    ebs_block_device {
      device_name  = "/dev/sdf"
      volume_size = 10
      volume_type = "gp2"
      delete_on_termination = true
      snapshot_id = "REDACTED"
    }
    root_block_device {
      delete_on_termination = true
      volume_size = 20
      volume_type = "gp2"
    }
}
```

**Comment by phinze** [comment]
Thanks for the report!

You'd think that the API would be cool with `encrypted: false` wouldn't you? But nope, it needs `nil`. :neutral_face: 

We'll get this fixed up.

**Comment by egarbi** [comment]
+1 for this one, it happened me today when I use trying to build a new "graphite" instance.

**Comment by amaczuga** [comment]
+1, occurs to me too. worked around by continuing apply with v0.4.2... (a disgusting way to do things)

**Comment by wkielas** [comment]
+1 from me, this effectively prevents from modifying root EBS volumes on instances

**Comment by netors** [comment]
+1 this happens to me if I try to use a snapshot of an encrypted volume

**Comment by sathiyas** [comment]
+1 from me, this effectively prevents from modifying root EBS volumes on instances

**Comment by catsby** [comment]
Patch for this in #2120

**Comment by catsby** [comment]
If anyone can build `master` and try #2120 out that would help me merge with confidence :smile: 

I took @randylyon config and was able to get through as expected, but other validation is always welcome

**Comment by armen** [comment]
I was going to send a patch for this and interestingly my patch is exactly same as yours, so I do confirm this works.

**Comment by catsby** [comment]
Closed by merging #2120 ,  let me know if this comes up again
